### DISCUSSION TITLE
Customized response message and status code in custom Lambda authorizer

### DISCUSSION THREAD

**Discussion by kkarthik-1** [aws/aws-lambda-dotnet#1816] [Q&A] [answered]
In our API Gateway, we are using a custom Lambda authorizer for authentication requests. The Lambda authorizer currently supports responses such as "Allow" and "Deny," which correspond to standard HTTP status codes like 401 (Unauthorized) and 403 (Forbidden).

However, we have a specific scenario where we need to return a 406 (Not Acceptable) response code for certain conditions, but Lambda authorizers seem to restrict us to only using 401 or 403 status codes for authentication failures.

Questions:

1. Is there any way to return a custom HTTP status code, like 406, from a custom Lambda authorizer in API Gateway? or i missed any things?

**Comment by bhoradc** [comment, marked as answer]
Hi @kkarthik-1, 

As per Amazon API Gateway [documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html), Lambda authorizers in API Gateway are limited to returning only 401 (Unauthorized) or 403 (Forbidden) status codes. It does not allow you to return custom HTTP status codes other than the standard 401 and 403 codes. 

Regards, 
Chaitanya
